src/webui/webapplication.cpp

```diff
@@ -320,7 +346,9 @@ void WebApplication::action_command_download()
             if ((url.size() == 40 && !url.contains(QRegExp("[^0-9A-Fa-f]")))
                 || (url.size() == 32 && !url.contains(QRegExp("[^2-7A-Za-z]"))))
                 url = "magnet:?xt=urn:btih:" + url;
-            BitTorrent::Session::instance()->addTorrent(url);
+
+            Net::DownloadManager::instance()->setCookiesFromUrl(cookies, QUrl::fromEncoded(url.toUtf8()));
```

Comment:
You can blame Qt for this.

The most common example is several torrent files from the same private tracker.  You'd think "well it's the same domain it should be the same cookie for all".  And you're right.

Sadly QNetworkCookie::setCookiesFromURL is "smart" and it asks for a whole URL instead of just the domain, and it adds the cookie to the cookieJar for the domain and "it did the work for you because you didn't need to parse the domain from the URL".  They actually say this with pride.

So basically if all the URLs are from the same domain, the first call will add it to the cookieJar and the next ones QNetworkCookie::setCookiesFromURL should notice it's already there for the same domain and just do nothing.